### PR TITLE
[ci][doc/6] find disrepancy between api in doc and code

### PR DESCRIPTION
**Pull request by can-anyscale**
Update the `check_api_discrepancy` to find issues in ray data API. If you check lints, it found some issues; but some are due to the current limitation of not understanding module alias (working on this in the next PR).

Test:
- CI

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
so we have currently zero discrepancy? how is a discrepancy introduced at the first place?

**Comment by can-anyscale** [comment]
there are many discrepancies found in the lint (but I don't make it fails): https://buildkite.com/ray-project/microcheck/builds/733#018fc19e-83a9-4380-8d48-f72458a38790/177-741. Some are false positive which I'm working on cleaning up in the next PRs.

File: ci/ray_ci/doc/cmd_check_api_discrepancy.py [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
+    print("\n")
+
+    # check that all public apis are in doc
+    print("Finding missing APIs in doc:\n")
+    for api_name, api_in_code in api_in_codes.items():
+        if (
+            api_in_code.annotation_type == AnnotationType.PUBLIC_API
+            and api_name not in api_in_docs
+        ):
+            print(f"\t{api_in_code}")
```
Comment (khluu) [inline ci/ray_ci/doc/cmd_check_api_discrepancy.py:37]: do we want to return 2 list of APIs instead of printing it out?

**Review by khluu — APPROVED** [review]

File: ci/ray_ci/doc/cmd_check_api_discrepancy.py (hunk 1 above)
Comment (khluu) [reply ci/ray_ci/doc/cmd_check_api_discrepancy.py:37]: ^ might be easier to parse for other work but also if you just want to take a look at which one doesn't meet the bar I think that's also fine

**Comment by can-anyscale** [comment]
let's run the lint check only when data api has been cleaned up

**Pull request closed without merging**